src/main/java/dukehandler/TaskManager.java

```diff
@@ -0,0 +1,212 @@
+            if ((tasks.get(i - 1).getStatusIcon()).equals(" ")) {
+                isAllDone = false;
+            }
+            System.out.println(" " + i + "." + tasks.get(i - 1).toString());
+        }
+        if (isAllDone) {
+            System.out.println(" Woah, all completed! Good job!");
+        }
+    }
+
+    public static void markTaskAsDone(String doneIndexString) {
+        int doneIndexInteger;
+        try {
+            doneIndexInteger = Integer.parseInt(doneIndexString);
+            checkTaskToMarkDone(doneIndexInteger);
+        } catch (ExceptionEmptyList eel) {
+            MessagePrinter.printEmptyListMessage();
+            return;
+        } catch (ExceptionIllegalTaskMarkedDone | NumberFormatException nfe) {
+            MessagePrinter.printDoneOrDeleteTaskErrorMessage("done", tasks.size());
+            return;
+        } catch (ExceptionAlreadyTaskMarkedDone atmd) {
+            System.out.println(" Hey, you've already marked that as done!");
+            return;
+        }
+
+        tasks.get(doneIndexInteger - 1).markAsDone();
+        System.out.println(" Nice! I've marked this task as done:\n "
+                + tasks.get(doneIndexInteger - 1).toString());
+    }
+
+    public static void checkTaskToMarkDone(int doneIndexInt)
+            throws ExceptionIllegalTaskMarkedDone,
+            ExceptionAlreadyTaskMarkedDone,
+            ExceptionEmptyList {
+        if (tasks.isEmpty()) {
+            throw new ExceptionEmptyList();
+        } else if (doneIndexInt < 1 || doneIndexInt > tasks.size()) {
+            throw new ExceptionIllegalTaskMarkedDone();
+        } else if ((tasks.get(doneIndexInt - 1).getStatusIcon()).equals("X")) {
+            throw new ExceptionAlreadyTaskMarkedDone();
+        }
+    }
+
+    public static void addNewTask(String taskType, String fullCommand) {
+        String[] part = fullCommand.split(taskType);
+        try {
+            checkNewTaskToAdd(taskType, part[1].trim());
+            Task newTask;
+            String description = part[1].trim();
+            String[] descriptionAndTime;
+            String taskName;
+            String timeOrDate;
+
+            switch (taskType) {
+            case "todo":
+                newTask = new ToDo(description);
+                tasks.add(newTask);
+                break;
+            case "deadline":
+                descriptionAndTime = description.split("/by");
+                taskName = descriptionAndTime[0].trim();
+                timeOrDate = descriptionAndTime[1].trim();
+                newTask = new Deadline(taskName, timeOrDate);
+                tasks.add(newTask);
+                break;
+            case "event":
+                descriptionAndTime = description.split("/at");
+                taskName = descriptionAndTime[0].trim();
+                timeOrDate = descriptionAndTime[1].trim();
+                newTask = new Event(taskName, timeOrDate);
+                tasks.add(newTask);
+                break;
```

Comment:
There seem to be a few similarities in the 2 case statements - try extracting them to reduce redundancy?